File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l, r = -1, 0
        visit = defaultdict(bool)
        
        while r < len(nums):
            if not visit[nums[r]]:
                l += 1
                nums[l] = nums[r]
                visit[nums[r]] = True
            r += 1
        
        return l + 1

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int s = 0; int f = 0;
        while (f < nums.length) {
            while (f < nums.length && nums[s] == nums[f]) f++;
            if (f < nums.length) {
                s++;
                nums[s] = nums[f];
            }
        }
        
        return s + 1;
    }
}